File: frontend/src/types/project-sub-status.ts
export interface ProjectSubStatus {
  id: string;
  name: string;
  sort_order: number;
  color: string;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface CreateProjectSubStatus {
  name: string;
  sort_order?: number;
  color?: string;
}

export interface UpdateProjectSubStatus {
  name?: string;
  sort_order?: number;
  color?: string;
  is_active?: boolean;
}

File: backend/crates/konto-db/src/entities/project_sub_status.rs
use sea_orm::entity::prelude::*;
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, PartialEq, DeriveEntityModel, Serialize, Deserialize)]
#[sea_orm(table_name = "project_sub_statuses")]
pub struct Model {
    #[sea_orm(primary_key, auto_increment = false)]
    pub id: String,
    pub name: String,
    pub sort_order: i32,
    pub color: String,
    pub is_active: bool,
    pub created_at: chrono::NaiveDateTime,
    pub updated_at: chrono::NaiveDateTime,
}

#[derive(Copy, Clone, Debug, EnumIter, DeriveRelation)]
pub enum Relation {}

impl ActiveModelBehavior for ActiveModel {}
